### PR TITLE
expression: handle max_allowed_packet warnings for repeat function.

### PR DESCRIPTION
**Pull request by hhxcc**
## What have you changed? (mandatory)
Return NULL and a warning when the result of repeat exceeds max_allowed_packetbs.

Before this PR:
```
tidb> SELECT repeat('h', 18446744073709551617);
+-----------------------------------+
| repeat('h', 18446744073709551617) |
+-----------------------------------+
| NULL                              |
+-----------------------------------+
1 row in set, 1 warning (0.00 sec)

tidb> show warnings;
+---------+------+-----------------------------------------------------------+
| Level   | Code | Message                                                   |
+---------+------+-----------------------------------------------------------+
| Warning | 1292 | Truncated incorrect DECIMAL value: '18446744073709551617' |
+---------+------+-----------------------------------------------------------+
1 row in set (0.00 sec)
```

After this PR:
```
tidb> SELECT repeat('h', 18446744073709551617);
+-----------------------------------+
| repeat('h', 18446744073709551617) |
+-----------------------------------+
| NULL                              |
+-----------------------------------+
1 row in set, 2 warnings (2.43 sec)

tidb> show warnings;
+---------+------+------------------------------------------------------------------------------+
| Level   | Code | Message                                                                      |
+---------+------+------------------------------------------------------------------------------+
| Warning | 1292 | Truncated incorrect DECIMAL value: '18446744073709551617'                    |
| Warning | 1301 | Result of repeat() was larger than max_allowed_packet (67108864) - truncated |
+---------+------+------------------------------------------------------------------------------+
2 rows in set (0.00 sec)
```

## What is the type of the changes? (mandatory)

Bug fix (non-breaking change which fixes an issue)

## How has this PR been tested? (mandatory)

- unit test
- explain test

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

## Does this PR need to be added to the release notes? (mandatory)
Yes

release note:
```
Return `NULL` when the result of function `REPEAT` is larger than `max_allowed_packet`

```

## Refer to a related PR or issue link (optional)
to [#7153](https://github.com/pingcap/tidb/issues/7153)

### CODE REVIEW

**Comment by zz-jason** [comment]
@hhxcc, #7171 has been merged into the master branch now, please merge master and resolve conflicts 😄

**Comment by hhxcc** [comment]
@zz-jason  ok, thanks for notify.

**Comment by zz-jason** [comment]
@hhxcc There are so many irrelevant changes, I think you may need to do some code clean work.

**Comment by hhxcc** [comment]
@zz-jason  sorry for that , I have already fixed it.

**Comment by zz-jason** [comment]
LGTM
@XuHuaiyu PTAL

File: expression/builtin_string.go [hunk 1]
```diff
@@ -575,6 +582,7 @@ func (b *builtinRepeatSig) evalString(row chunk.Row) (d string, isNull bool, err
 	if isNull || err != nil {
 		return "", isNull, errors.Trace(err)
 	}
+	byteLength := len(str)
```
Comment (XuHuaiyu) [inline expression/builtin_string.go]: It seems that this is rune length, not byte length?
I think we need to add some test cased for multi-byte characters, like Chinese character.

Comment (hhxcc) [reply expression/builtin_string.go]: Len of String will return the byte length,  you can check it in  https://golang.org/pkg/builtin/#len .
I have added some test of Chinese character.

File: expression/builtin_string_test.go [hunk 2]
```diff
@@ -398,6 +398,53 @@ func (s *testEvaluatorSuite) TestRepeat(c *C) {
 	c.Assert(v.GetString(), Equals, "")
 }
 
+func (s *testEvaluatorSuite) TestRepeatSig(c *C) {
```
Comment (crazycs520) [inline expression/builtin_string_test.go]: Could you refine the test? 
Maybe you can keep style with another tests.

**Comment by zz-jason** [comment]
@hhxcc It would be great if you can cherrypick this PR to the `release-2.0` 😄

**Comment by hhxcc** [comment]
@zz-jason  ok

**Pull request merged**